src/Components/Fsi.fs

```diff
@@ -494,19 +494,40 @@ module Fsi =
             return terminal
         }
 
+    let detectTerminalSize (terminal: Terminal) =
+        promise {
+            let! terminalId = Promise.ofThenable terminal.processId
+
+            match terminalId with
+            | Some id ->
+                let! terminalSize = execCommand "stty" [ "size"; "-F"; sprintf "/proc/%d/fd/0" id ]
+                let rows, cols = terminalSize.Split(' ') |> Array.map int |> fun arr -> arr.[0], arr.[1]
+                return Some(rows, cols)
+            | None -> return None
+        }
```

Comment:
not sure if this part will help actually (?) gen by copilot ws, does not look right